### PR TITLE
Kurtwheeler/crunch rerun fixes

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

https://github.com/AlexsLemonade/refinebio/issues/182

## Purpose/Implementation Notes

This fixes a few bugs that I came across in the process of rerunning the Neuroblastoma crunch for #182. After I fixed those bugs, I wanted to make a test to keep them squashed so I built out the infrastructure to run end-to-end tests which is included in this PR.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Functional tests

I've rerun the experiments that caused the bugs to appear and verified that at least the surveyor was able to complete successfully. (The downloaders and processors are a lot more processing.)

I've also made sure the tests pass locally and in CircleCI

## Checklist

_Put an `x` in the boxes that apply._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: common/requirements.txt [hunk 1]
```diff
@@ -15,8 +15,7 @@ django==2.0.2
 docutils==0.14            # via botocore
 idna==2.6                 # via requests
 jmespath==0.9.3           # via boto3, botocore
-pip==9.0.3
```
Comment (Miserlou) [inline common/requirements.txt]: Why did you remove pinning pip? Are we sure we're entirely pip10 compatible?

File: foreman/data_refinery_foreman/surveyor/management/commands/clear_database.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"""
+This command will clear out the database to make repeating tests easier.
+"""
+
+from django.core.management.base import BaseCommand
+from data_refinery_common.models import *
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        Sample.objects.all().delete()
```
Comment (Miserlou) [inline foreman/data_refinery_foreman/surveyor/management/commands/clear_database.py]: This is a pretty dangerous function, maybe get user input for a (Y/n)?

File: common/requirements.txt (hunk 1 above)
Comment (kurtwheeler) [reply common/requirements.txt]: I can't think of any way we wouldn't be, all the images build successfully and the tests pass.

I figure we've gotta remove it at some point, we can't stay on that version forever.

**Comment by Miserlou** [comment]
I'm getting some failures when trying to `run_all_tests` on this branch:

```
[ .. ]
Sending build context to Docker daemon  7.743MB
Step 1/10 : FROM python:3.6.1-slim
 ---> 93b738c915e5
Step 2/10 : RUN groupadd user && useradd --create-home --home-dir /home/user -g user user
 ---> Using cache
 ---> dde05a3ae12e
Step 3/10 : WORKDIR /home/user
 ---> Using cache
 ---> c331b3c5b345
Step 4/10 : COPY foreman/requirements.txt .
 ---> Using cache
 ---> ad6f5e6649f0
Step 5/10 : RUN pip install -r requirements.txt
 ---> Using cache
 ---> 8793a65ef0e0
Step 6/10 : COPY common/dist/data-refinery-common-* common/
COPY failed: no source files were specified
/usr/local/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
Unknown command: 'bash'
Type 'manage.py help' for usage.
Downloading ERR003000_1.fastq.gz for Salmon tests.
Downloading ERR003000_2.fastq.gz for Salmon tests.
[ .. ]
```

and later:

```
System check identified no issues (0 silenced).
EEEEEEEE
======================================================================
ERROR: data_refinery_workers.downloaders.test_array_express (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.downloaders.test_array_express
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/downloaders/test_array_express.py", line 5, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.downloaders.test_sra (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.downloaders.test_sra
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/downloaders/test_sra.py", line 5, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.downloaders.test_transcriptome_index (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.downloaders.test_transcriptome_index
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/downloaders/test_transcriptome_index.py", line 6, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.processors.test_array_express (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.processors.test_array_express
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/processors/test_array_express.py", line 6, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.processors.test_no_op (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.processors.test_no_op
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/processors/test_no_op.py", line 4, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.processors.test_salmon (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.processors.test_salmon
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/processors/test_salmon.py", line 5, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.processors.test_transcriptome_index (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.processors.test_transcriptome_index
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/processors/test_transcriptome_index.py", line 6, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


======================================================================
ERROR: data_refinery_workers.processors.test_utils (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: data_refinery_workers.processors.test_utils
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/user/data_refinery_workers/processors/test_utils.py", line 5, in <module>
    from data_refinery_common.models import (
ImportError: cannot import name 'Batch'


----------------------------------------------------------------------
Ran 8 tests in 0.001s
```

**Comment by kurtwheeler** [comment]
Ok, so I've addressed both your comments. Anything else?

**Comment by Miserlou** [comment]
Test suite runs locally for me, assuming these (swallowed) errors are okay:

```
2018-05-02 21:19:06,667 local/MainProcess data_refinery_workers.processors.utils ERROR [processor_job: 6]: Unhandled exception caught while running processor _download_index in pipeline
Traceback (most recent call last):
  File "/home/user/data_refinery_workers/processors/utils.py", line 164, in run_pipeline
    last_result = processor(last_result)
  File "/home/user/data_refinery_workers/processors/salmon.py", line 157, in _download_index
    tarball.extractall(job_context["index_unpacked"])
  File "/usr/lib/python3.5/tarfile.py", line 1996, in extractall
    numeric_owner=numeric_owner)
  File "/usr/lib/python3.5/tarfile.py", line 2038, in extract
    numeric_owner=numeric_owner)
  File "/usr/lib/python3.5/tarfile.py", line 2108, in _extract_member
    self.makefile(tarinfo, targetpath)
  File "/usr/lib/python3.5/tarfile.py", line 2156, in makefile
    copyfileobj(source, target, tarinfo.size, ReadError)
  File "/usr/lib/python3.5/tarfile.py", line 241, in copyfileobj
    buf = src.read(BUFSIZE)
  File "/usr/lib/python3.5/gzip.py", line 274, in read
    return self._buffer.read(size)
  File "/usr/lib/python3.5/_compression.py", line 68, in readinto
    data = self.read(len(byte_view))
  File "/usr/lib/python3.5/gzip.py", line 480, in read
    raise EOFError("Compressed file ended before the "
EOFError: Compressed file ended before the end-of-stream marker was reached
```

**Comment by Miserlou** [comment]
Noting for reference: Ensembl has been down all day and this looks like it's index-related, however maybe this should have failed? Missing an assertion maybe?
